tutorials/hyperspectral/Advanced_Spectral_Signatures.Rmd

```diff
@@ -0,0 +1,291 @@
+
+### Recommended Skills
+
+We highly recommend you work through the 
+<a href="https://neonscience.org/hsi-hdf5-r" target="_blank"> *Introduction to Working with Hyperspectral Data in HDF5 Format in R* tutorial</a>
+before moving on to this tutorial.
+
+</div>
+
+
+## Import libraries and load datasets
+First, we need to load our required libraries, and import the hyperspectral
+data (in HDF5 format).
+
+```{r load-libraries, message=FALSE, warning=FALSE}
+
+# Load required packages
+library(rhdf5)
+library(reshape2)
+library(raster)
+library(plyr)
+library(ggplot2)
+
+# set working directory to ensure R can find the file we wish to import and where
+# we want to save our files. Be sure to move the download into your working directory!
+wd="~/Desktop/Hyperspectral_Tutorial/" #This will depend on your local environment
+setwd(wd)
+
+# define filepath to the hyperspectral dataset
+f <- paste0(wd,"NEON_hyperspectral_tutorial_example_subset.h5")
+
+# read in the wavelength information from the HDF5 file
+wavelengths <- h5read(f,"/SJER/Reflectance/Metadata/Spectral_Data/Wavelength")
+
+# grab scale factor from the Reflectance attributes
+reflInfo<- h5readAttributes(f,"/SJER/Reflectance/Reflectance_Data" )
+
+scaleFact = reflInfo$Scale_Factor
+
+```
+
+Now, we read in the RGB image that we created in an earlier tutorial and plot it.
+
+```{r read-in-RGB-and-plot}
+
+# Read in RGB image as a 'stack' rather than a plain 'raster'
+rgbStack <- stack(paste0(wd,"NEON_hyperspectral_tutorial_example_RGB_stack_image.tif"))
+
+## Plot as RGB image
+plotRGB(rgbStack,
+        r=1,g=2,b=3, scale=300, 
+        stretch = "lin")
+
+```
+
+## Interactive `click` function from `raster` package
+Next, we use an interactive clicking function to identify the pixels that we want
+to extract spectral signatures for. To follow along best with this tutorial, we 
+suggest the following five cover types (exact location shown below). 
+1. Irrigated grass
+2. Tree canopy (avoid the shaded northwestern side of the tree)
+3. Roof
+4. Bare soil (infield)
+5. Open water
+
+As shown here:
+<figure >
+    <a href="{{ site.baseurl }}/images/hyperspectral/Click_points.png"><img src="{{ site.baseurl }}/images/hyperspectral/Click_points.png"></a>
+<figcaption> Five different land cover types chosen for this study (magenta dots) in the order listed above (red numbers).</figcaption>
+</figure>
+
+
+```{r click-to-select, eval=FALSE, comment=NA}
+
+# change plotting parameters to better see the points and numbers generated from clicking
+par(col="red", cex=3)
+
+# use the 'click' function
+c=click(rgbStack, id=T, xy=T, cell=T, type="p", pch=16, col="magenta", col.lab="red")
+
+```
+
+Once you have clicked your five points, press the `ESC` key to save your
+clicked points and close the function before moving on to the next step. If 
+you make a mistake in the step, run the `plotRGB` function again to start over.
+```{r create-c-hidden, include=FALSE, purl=FALSE}
```

Comment:
I agree, and usually do. This is the only chunk where I didn't do that, because it is hidden using the include=FALSE, purl=FALSE so it will never show up in the .md or .R file. By not including a blank line above, I am avoiding a double-wide gap between the preceding and subsequent paragraphs.